extensions/amp-analytics/0.1/test/test-amp-analytics.js

```diff
@@ -250,18 +250,47 @@ describes.realWin('amp-analytics', {
     }
   });
 
-  it('preload script', function() {
+  it('does not unnecessarily preload iframe transport script', function() {
     const el = doc.createElement('amp-analytics');
     doc.body.appendChild(el);
     const analytics = new AmpAnalytics(el);
+    sandbox.stub(analytics, 'assertAmpAdResourceId').callsFake(() => 'fakeId');
+    const preloadSpy = sandbox.spy(analytics, 'preload');
     analytics.buildCallback();
     analytics.preconnectCallback();
-    return Promise.resolve().then(() => {
-      const preloads = doc.querySelectorAll('link[rel=preload]');
-      expect(preloads).to.have.length(1);
-      expect(preloads[0]).to.have.property(
-          'href',
-          'http://localhost:9876/dist/iframe-transport-client-lib.js');
+    return analytics.layoutCallback().then(() => {
+      expect(preloadSpy).to.have.not.been.called;
+    });
+  });
+
+  it('preloads iframe transport script if relevant', function() {
+    const el = doc.createElement('amp-analytics');
+    el.setAttribute('type', 'foo');
+    doc.body.appendChild(el);
+    const analytics = new AmpAnalytics(el);
+    sandbox.stub(analytics, 'assertAmpAdResourceId').callsFake(() => 'fakeId');
+    const preloadSpy = sandbox.spy(analytics, 'preload');
+    analytics.predefinedConfig_['foo'] = {
+      'transport': {
+        'iframe': 'https://www.google.com',
+      },
+      'triggers': {
+        'sample_visibility_trigger': {
+          'on': 'visible',
+          'request': 'sample_visibility_request',
+        },
+      },
+      'requests': {
+        'sample_visibility_request': 'fake-request',
+      },
+    };
+    analytics.buildCallback();
+    analytics.preconnectCallback();
+    return analytics.layoutCallback().then(() => {
+      expect(preloadSpy).to.have.been.calledOnce;
```

Comment:
FYI you can actually combine these by doing expect(preloadSpy.withArgs(...)).to.be.calledOnce;